File: Code/JavaScript/client/src/messages/HVPSStateChangedMessage.js
export default class HVPSStateChangedMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="sc";
    r["s"]=o.state;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.state=o["s"];
   return r;
 }
}

File: Code/C#/HVPSMessages/Messages/HVPSStateChangedMessage.cs
using Core.Messages.Messages;
using HVPSCore.Enums;
using HVPSMessages.DataMemberNames.Messages;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace HVPSMessages.Messages
{
    [DataContract]
    public class HVPSStateChangedMessage : TypedMessageBase
    {
        [JsonPropertyName(HVPSStateChangedMessageDataMemberNames.State)]
        [JsonInclude]
        [DataMember(Name = HVPSStateChangedMessageDataMemberNames.State)]
        public HVPSState State { get; protected set; }
        public HVPSStateChangedMessage()
            : base()
        {
            Type = MessageTypes.HVPSUIStateChanged;
        }
    }
}
